services/QuillLMS/spec/models/unit_activity_spec.rb

```diff
@@ -146,19 +142,66 @@
 
   context 'self.get_classroom_user_profile' do
     it 'get user profile data' do
-      units = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
-      expect(units.count).to eq(2)
+      unit_activities = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
+      expect(unit_activities.count).to eq(2)
     end
 
     it 'gracefully account for nils' do
-      units = UnitActivity.get_classroom_user_profile(nil, nil)
-      expect(units.count).to eq(0)
+      unit_activities = UnitActivity.get_classroom_user_profile(nil, nil)
+      expect(unit_activities.count).to eq(0)
     end
 
-    it 'includes units even if their activities are archived' do
+    it 'includes unit_activities even if their activities are archived' do
       activity.update(flags: [Activity::ARCHIVED])
-      units = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
-      expect(units.count).to eq(2)
+      unit_activities = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
+      expect(unit_activities.count).to eq(2)
     end
+
+    describe 'publish dates' do
+
+      it 'includes unit activities that do not have a publish date' do
+        unit_activity.update(publish_date: nil)
+        lessons_unit_activity.update(publish_date: nil)
+        unit_activities = UnitActivity.get_classroom_user_profile(classroom.id, student.id)
+        expect(unit_activities.count).to eq(2)
```

Comment:
The `count` is a fine thing to check here, but it may also be worth checking that the expected values are in the results.  That's a bit tricky here because of the use of raw SQL, but you could maybe do something like:
```
expect(unit_activities.map{ |ua| ua['ua_id'] }).to include(unit_activity.id, lessons_unit_activity.id)
```
If that makes sense, it may be worth adding similar conditions to the follow-on cases as well.